guava/src/com/google/common/util/concurrent/Uninterruptibles.java

```diff
@@ -326,7 +326,67 @@ public static boolean tryAcquireUninterruptibly(
     }
   }
 
-  // TODO(user): Add support for waitUninterruptibly.
+  /**
+   * Invokes {@code object.}{@link Object#wait() wait()} uninterruptibly.
+   *
+   * @param object to wait upon.
+   *
+   * @throws IllegalMonitorStateException if the current thread is not
+   *               the owner of the object's monitor.
+   */
+  @GwtIncompatible // concurrency
+  public static void waitUninterruptibly(Object object) {
+    Preconditions.checkNotNull(object);
+    boolean interrupted = false;
+    try {
+      while (true) {
+        try {
+          object.wait();
+          return;
+        } catch (InterruptedException e) {
+          interrupted = true;
```

Comment:
http://docs.oracle.com/javase/specs/jls/se8/html/jls-17.html#jls-17.2.1

And for my reference:
- http://b/4198432#comment4
- http://cl/19874683